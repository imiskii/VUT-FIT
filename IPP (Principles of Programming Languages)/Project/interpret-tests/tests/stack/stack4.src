.ippcode23

DEFVAR GF@a
DEFVAR GF@b
DEFVAR GF@x

MOVE GF@a int@5
MOVE GF@b float@+3.14

PUSHS int@5
PUSHS int@100
PUSHS float@-3.14
PUSHS float@42.42

ADDS
POPS GF@x
WRITE GF@x

SUBS
POPS GF@x
WRITE GF@x

PUSHS int@5
PUSHS int@100
PUSHS float@-3.14
PUSHS float@42.42

IDIVS
POPS GF@x
WRITE GF@x

MULS
POPS GF@x
WRITE GF@x

PUSHS int@100
PUSHS GF@a
PUSHS float@-3.14
PUSHS GF@b

SUBS
POPS GF@x
WRITE GF@x

IDIVS
POPS GF@x
WRITE GF@x