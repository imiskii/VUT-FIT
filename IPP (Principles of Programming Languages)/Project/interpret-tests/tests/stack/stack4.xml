<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode23">
 <instruction order="1" opcode="DEFVAR">
  <arg1 type="var">GF@a</arg1>
 </instruction>
 <instruction order="2" opcode="DEFVAR">
  <arg1 type="var">GF@b</arg1>
 </instruction>
 <instruction order="3" opcode="DEFVAR">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="4" opcode="MOVE">
  <arg1 type="var">GF@a</arg1>
  <arg2 type="int">5</arg2>
 </instruction>
 <instruction order="5" opcode="MOVE">
  <arg1 type="var">GF@b</arg1>
  <arg2 type="float">+3.14</arg2>
 </instruction>
 <instruction order="6" opcode="PUSHS">
  <arg1 type="int">5</arg1>
 </instruction>
 <instruction order="7" opcode="PUSHS">
  <arg1 type="int">100</arg1>
 </instruction>
 <instruction order="8" opcode="PUSHS">
  <arg1 type="float">-3.14</arg1>
 </instruction>
 <instruction order="9" opcode="PUSHS">
  <arg1 type="float">42.42</arg1>
 </instruction>
 <instruction order="10" opcode="ADDS"/>
 <instruction order="11" opcode="POPS">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="12" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="13" opcode="SUBS"/>
 <instruction order="14" opcode="POPS">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="15" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="16" opcode="PUSHS">
  <arg1 type="int">5</arg1>
 </instruction>
 <instruction order="17" opcode="PUSHS">
  <arg1 type="int">100</arg1>
 </instruction>
 <instruction order="18" opcode="PUSHS">
  <arg1 type="float">-3.14</arg1>
 </instruction>
 <instruction order="19" opcode="PUSHS">
  <arg1 type="float">42.42</arg1>
 </instruction>
 <instruction order="20" opcode="IDIVS"/>
 <instruction order="21" opcode="POPS">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="22" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="23" opcode="MULS"/>
 <instruction order="24" opcode="POPS">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="25" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="26" opcode="PUSHS">
  <arg1 type="int">100</arg1>
 </instruction>
 <instruction order="27" opcode="PUSHS">
  <arg1 type="var">GF@a</arg1>
 </instruction>
 <instruction order="28" opcode="PUSHS">
  <arg1 type="float">-3.14</arg1>
 </instruction>
 <instruction order="29" opcode="PUSHS">
  <arg1 type="var">GF@b</arg1>
 </instruction>
 <instruction order="30" opcode="SUBS"/>
 <instruction order="31" opcode="POPS">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="32" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="33" opcode="IDIVS"/>
 <instruction order="34" opcode="POPS">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="35" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
</program>
