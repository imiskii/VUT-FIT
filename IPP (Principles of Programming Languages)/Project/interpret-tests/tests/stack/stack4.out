39.28-95-0.07402168788307402500-6.2820.0